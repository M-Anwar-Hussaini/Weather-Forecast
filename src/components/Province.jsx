import WeatherIcon from 'react-open-weather-icons';
import Type from 'prop-types';
import { FaLocationDot, FaTemperatureArrowUp } from 'react-icons/fa6';

export default function Province({
  province, icon, temprature, className, description,
}) {
  return (
    <figure className={`p-3 m-0 ${className}`} id="province">
      <div className="row align-items-center justify-content-between">
        <div className="col-6 d-flex flex-column gap-3">
          <div>
            <div className="d-flex gap-2 align-items-center">
              <FaLocationDot fontSize={24} />
              <span className="fs-6 fw-bold">{province}</span>
            </div>
          </div>

          <div className="d-flex align-items-center gap-2">
            <FaTemperatureArrowUp fontSize={24} />
            <span>
              {temprature}
              &deg;C
            </span>
          </div>
        </div>
        <div className="col-6 d-flex flex-column align-items-center">
          <WeatherIcon name={icon} className="weather-icon" />
          <span>{description}</span>
        </div>

      </div>
    </figure>
  );
}

Province.propTypes = {
  province: Type.string.isRequired,
  icon: Type.string.isRequired,
  temprature: Type.number.isRequired,
  className: Type.string.isRequired,
  description: Type.string.isRequired,
};
