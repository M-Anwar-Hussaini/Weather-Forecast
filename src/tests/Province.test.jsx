import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Province from '../components/Province';

const province = (
  <Province className="" description="Kabul" icon="01d" province="Kabul" temprature={23} />
);
describe('Apply tests for Province component', () => {
  test('Take snapshot from Province', () => {
    const pro = render(province);
    expect(pro).toMatchSnapshot();
  });

  test('Province has a figure element', () => {
    render(province);
    const figure = screen.getByRole('figure');
    expect(figure).toBeInTheDocument();
  });

  test('figue element has one direct child', () => {
    render(province);
    const figure = screen.getByRole('figure');
    expect(figure.children).toHaveLength(1);
  });

  test('Wrapper component has 2 children', () => {
    render(province);
    const figure = screen.getByRole('figure');
    expect(figure.children[0].children).toHaveLength(2);
  });
});
