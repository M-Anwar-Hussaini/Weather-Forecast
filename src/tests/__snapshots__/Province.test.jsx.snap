// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply tests for Province component Take snapshot from Province 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <figure
        class="h-100 p-3 m-0 "
        id="province"
      >
        <div
          class="row align-items-center justify-content-between"
        >
          <div
            class="col-7 d-flex flex-column gap-3"
          >
            <div>
              <div
                class="d-flex gap-2 align-items-center"
              >
                <svg
                  fill="currentColor"
                  font-size="24"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                  />
                </svg>
                <span
                  class="fs-6 fw-bold"
                >
                  Kabul
                </span>
              </div>
            </div>
            <div
              class="d-flex align-items-center gap-2"
            >
              <svg
                fill="currentColor"
                font-size="24"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 112c0-26.5 21.5-48 48-48s48 21.5 48 48V276.5c0 17.3 7.1 31.9 15.3 42.5C249.8 332.6 256 349.5 256 368c0 44.2-35.8 80-80 80s-80-35.8-80-80c0-18.5 6.2-35.4 16.7-48.9c8.2-10.6 15.3-25.2 15.3-42.5V112zM176 0C114.1 0 64 50.1 64 112V276.4c0 .1-.1 .3-.2 .6c-.2 .6-.8 1.6-1.7 2.8C43.2 304.2 32 334.8 32 368c0 79.5 64.5 144 144 144s144-64.5 144-144c0-33.2-11.2-63.8-30.1-88.1c-.9-1.2-1.5-2.2-1.7-2.8c-.1-.3-.2-.5-.2-.6V112C288 50.1 237.9 0 176 0zm0 416c26.5 0 48-21.5 48-48c0-20.9-13.4-38.7-32-45.3V112c0-8.8-7.2-16-16-16s-16 7.2-16 16V322.7c-18.6 6.6-32 24.4-32 45.3c0 26.5 21.5 48 48 48zM480 160h32c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-64-64c-12.5-12.5-32.8-12.5-45.3 0l-64 64c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8h32V448c0 17.7 14.3 32 32 32s32-14.3 32-32V160z"
                />
              </svg>
              <span>
                23
                °C
              </span>
            </div>
          </div>
          <div
            class="col-5 d-flex flex-column align-items-center"
          >
            <img
              class="weather-icon"
              src="wi-day-sunny.svg"
            />
            <span>
              Kabul
            </span>
          </div>
        </div>
      </figure>
    </div>
  </body>,
  "container": <div>
    <figure
      class="h-100 p-3 m-0 "
      id="province"
    >
      <div
        class="row align-items-center justify-content-between"
      >
        <div
          class="col-7 d-flex flex-column gap-3"
        >
          <div>
            <div
              class="d-flex gap-2 align-items-center"
            >
              <svg
                fill="currentColor"
                font-size="24"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M215.7 499.2C267 435 384 279.4 384 192C384 86 298 0 192 0S0 86 0 192c0 87.4 117 243 168.3 307.2c12.3 15.3 35.1 15.3 47.4 0zM192 128a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                />
              </svg>
              <span
                class="fs-6 fw-bold"
              >
                Kabul
              </span>
            </div>
          </div>
          <div
            class="d-flex align-items-center gap-2"
          >
            <svg
              fill="currentColor"
              font-size="24"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128 112c0-26.5 21.5-48 48-48s48 21.5 48 48V276.5c0 17.3 7.1 31.9 15.3 42.5C249.8 332.6 256 349.5 256 368c0 44.2-35.8 80-80 80s-80-35.8-80-80c0-18.5 6.2-35.4 16.7-48.9c8.2-10.6 15.3-25.2 15.3-42.5V112zM176 0C114.1 0 64 50.1 64 112V276.4c0 .1-.1 .3-.2 .6c-.2 .6-.8 1.6-1.7 2.8C43.2 304.2 32 334.8 32 368c0 79.5 64.5 144 144 144s144-64.5 144-144c0-33.2-11.2-63.8-30.1-88.1c-.9-1.2-1.5-2.2-1.7-2.8c-.1-.3-.2-.5-.2-.6V112C288 50.1 237.9 0 176 0zm0 416c26.5 0 48-21.5 48-48c0-20.9-13.4-38.7-32-45.3V112c0-8.8-7.2-16-16-16s-16 7.2-16 16V322.7c-18.6 6.6-32 24.4-32 45.3c0 26.5 21.5 48 48 48zM480 160h32c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-64-64c-12.5-12.5-32.8-12.5-45.3 0l-64 64c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8h32V448c0 17.7 14.3 32 32 32s32-14.3 32-32V160z"
              />
            </svg>
            <span>
              23
              °C
            </span>
          </div>
        </div>
        <div
          class="col-5 d-flex flex-column align-items-center"
        >
          <img
            class="weather-icon"
            src="wi-day-sunny.svg"
          />
          <span>
            Kabul
          </span>
        </div>
      </div>
    </figure>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
